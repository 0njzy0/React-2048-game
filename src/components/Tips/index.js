import React from 'react';
import PropTypes from 'prop-types';
import styles from './tips.scss';

export default function Tips({ children }) {
  return (
    <div className={styles.tips}>
      {/* eslint-disable react/no-danger */}
      <p className={styles.content} dangerouslySetInnerHTML={{ __html: children }} />
    </div>
  );
}

Tips.propTypes = {
  children: PropTypes.node.isRequired,
};
