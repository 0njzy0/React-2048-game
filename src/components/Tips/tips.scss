@import '../../assets/styles/vars';

.tips {
  background: $gray;
  border-radius: 5px;
  color: $black;
  font-size: 16px;
  margin: 50px auto 0;
  width: 570px;

  .content {
    padding: 15px 0;
  }
}
