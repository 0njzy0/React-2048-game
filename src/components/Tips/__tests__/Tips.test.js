import React from 'react';
import renderer from 'react-test-renderer';
import Tips from '..';

describe('<Tips />', () => {
  it('component render', () => {
    let tips = renderer.create(
      <Tips>
        test
      </Tips>,
    ).toJSON();
    tips = renderer.create(
      <Tips>
        <p>hello <span>world</span></p>
      </Tips>,
    ).toJSON();
    expect(tips).toMatchSnapshot();
  });
});
