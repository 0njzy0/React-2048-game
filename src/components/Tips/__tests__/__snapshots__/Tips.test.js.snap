// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tips /> component render 1`] = `
<div
  className="tips"
>
  <p
    className="content"
    dangerouslySetInnerHTML={
      Object {
        "__html": <p>
          hello 
          <span>
            world
          </span>
        </p>,
      }
    }
  />
</div>
`;
